src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/illegaltype/InputIllegalTypeRecordsWithMemberModifiersDefault.java

```diff
@@ -0,0 +1,63 @@
+/*
+IllegalType
+validateAbstractClassNames = (default)false
+illegalClassNames = (default)HashMap, HashSet, LinkedHashMap, LinkedHashSet, TreeMap, TreeSet, \
+                    java.util.HashMap, java.util.HashSet, java.util.LinkedHashMap, \
+                    java.util.LinkedHashSet, java.util.TreeMap, java.util.TreeSet
+legalAbstractClassNames = (default)
+ignoredMethodNames = (default)getEnvironment, getInitialContext
+illegalAbstractClassNameFormat = (default)^(.*[.])?Abstract.*$
+memberModifiers = (default)
+tokens = (default)ANNOTATION_FIELD_DEF, CLASS_DEF, INTERFACE_DEF, METHOD_CALL, METHOD_DEF, \
+         METHOD_REF, PARAMETER_DEF, VARIABLE_DEF, PATTERN_VARIABLE_DEF, RECORD_DEF, \
+         RECORD_COMPONENT_DEF
+
+
+*/
+
+//non-compiled with javac: Compilable with Java14
+package com.puppycrawl.tools.checkstyle.checks.coding.illegaltype;
+
+import java.util.*;
+
+public class InputIllegalTypeRecordsWithMemberModifiersDefault
+{
+
+    public record IdentifiersPair(
+        UUID productId, // ok
+        String identifier // ok
+    )
+    {
+
+    }
+
+    public class IdentifiersPairEquivalent {
+        private final UUID productId; // ok
+        private final String identifier; // ok
+
+        public IdentifiersPairEquivalent(UUID productId, String identifier) {
+            this.productId = productId;
+            this.identifier = identifier;
+        }
+    }
+
+    public record IdentifiersPair2(
+        HashSet x, // ok
```

Comment:
It should be violation.
Why is it ok ?

Reply:
I first made it violation but I was getting an error that it is not violation with the given condition

Reply:
![44](https://user-images.githubusercontent.com/71710042/144758631-57bb0ba9-bc30-4f26-9b0a-9b58e8731ba3.png)

Reply:
Please update expected list in Test.
It should be violation.

Reply:
45 is an expected list but it is still showing that it is not an error. I added the expected List in the Test but still it says that it shouldn't be the violation for this line.

Reply:
![aff](https://user-images.githubusercontent.com/71710042/144861831-e48b638a-bae4-44d5-a6f8-e9ca2e6b9832.png)

Reply:
I updated the ExpectedList in Test and also added  `HashSet x, // violation`
But I am still getting an error.

Reply:
@nmancus1 can you please help me with this?

Reply:
Please see https://github.com/checkstyle/checkstyle/pull/10958#discussion_r762661072 

You need to make changes in the check to align with behavior specified by @romani at https://github.com/checkstyle/checkstyle/pull/10958#discussion_r762466240.

Reply:
@nmancus1 like this?
```
 private void visitRecordComponentDef(DetailAST recordComponentDef) {
        if (memberModifiers.contains(TokenTypes.RECORD_COMPONENT_DEF)) {
            checkClassName(recordComponentDef);
        }
    }
```